lib/Model/Document/TypeDefinition/Loader/TypeLoader.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types = 1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Model\Document\TypeDefinition\Loader;
+
+use Pimcore\Loader\ImplementationLoader\ImplementationLoader;
+use Pimcore\Model\Document;
+
+/**
+ * @internal
+ */
+class TypeLoader extends ImplementationLoader implements TypeLoaderInterface
```

Comment:
I think we can use `ImplementationLoader` here directly right, then we also don't need the dedicated `TypeLoaderInterface`, right?